benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -0,0 +1,149 @@
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    prelude::{Added, Changed},
+    world::World,
+};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::{prelude::SliceRandom, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+criterion_group!(
+    benches,
+    all_added_detection,
+    all_changed_detection,
+    few_changed_detection,
+);
+criterion_main!(benches);
+
+#[derive(Component, Default)]
+#[component(storage = "Table")]
+struct Table;
+#[derive(Component, Default)]
+#[component(storage = "SparseSet")]
+struct Sparse;
+
+const RANGE_ENTITIES_TO_BENCH_COUNT: std::ops::Range<u32> = 5..7;
+
+type BenchGroup<'a> = criterion::BenchmarkGroup<'a, criterion::measurement::WallTime>;
+
+fn deterministic_rand() -> ChaCha8Rng {
+    ChaCha8Rng::seed_from_u64(42)
+}
+
+fn setup<T: Component + Default>(entity_count: u32) -> World {
+    let mut world = World::default();
+    world.spawn_batch((0..entity_count).map(|_| (T::default(),)));
+    black_box(world)
+}
+
+#[macro_export]
+macro_rules! bevy_bench {
+    ( $( $bench:ident, $harness:ident),+ ) => {
+        $(
+            fn $harness(criterion: &mut Criterion) {
+                let mut group: criterion::BenchmarkGroup<criterion::measurement::WallTime> =
+                    criterion.benchmark_group(stringify!($bench));
+                group.warm_up_time(std::time::Duration::from_millis(500));
+                group.measurement_time(std::time::Duration::from_secs(4));
+
+                for entity_count in RANGE_ENTITIES_TO_BENCH_COUNT.map(|i| i * 10_000) {
+                    $bench::<Table>(&mut group, entity_count);
+                    $bench::<Sparse>(&mut group, entity_count);
+                }
+
+                group.finish();
+            }
+        )+
+    };
+}
```

Comment:
This goes clearly beyond the goal of this PR, but I felt like it's a good way to remove duplication, I can do that in a separate PR if it's more appropriate

Reply:
I went for a generic function rather than a macro.